tests/test_resolver.py

```diff
@@ -827,3 +828,26 @@ def test_resolve_from_pex_ignore_errors(
     assert len(resolved_distributions_by_key) > 1, "We should resolve at least requests and urllib3"
     assert "requests" in resolved_distributions_by_key
     assert Requirement.parse("urllib3==1.26.1") == resolved_distributions_by_key["urllib3"]
+
+
+def test_pip_proprietary_url_with_markers_issues_1415():
+    # https://<internal hostname>/pip/ray-0.8.4-cp37-cp37m-linux_x86_64.whl; sys_platform == 'linux'
+    # ray==0.8.4; sys_platform == 'darwin'
+    resolved_dists = resolve_multi(
```

Comment:
Prior to the fix, this failed much like the OP in #1415:
```console
E               AssertionError: More than one direct requirement is satisfied by ansicolors 1.1.8:
E               0. ansicolors==1.1.8
E               1. ansicolors==1.1.8; sys_platform == "linux"
E               This should never happen since Pip fails when more than one requirement for a given project name key is supplied and applies for a given target interpreter environment.
```